src/api/schema/metrics.rs

```diff
@@ -1,11 +1,19 @@
 use crate::event::{Event, Metric, MetricValue};
-use crate::metrics::{capture_metrics, get_controller};
+use crate::metrics::{capture_metrics, get_controller, Controller};
 use async_graphql::{validators::IntRange, Interface, Object, Subscription};
 use async_stream::stream;
 use chrono::{DateTime, Utc};
+use lazy_static::lazy_static;
+use std::sync::{Arc, Mutex};
 use tokio::stream::{Stream, StreamExt};
 use tokio::time::Duration;
 
+lazy_static! {
+    static ref GLOBAL_CONTROLLER: Arc<Mutex<&'static Controller>> = Arc::new(Mutex::new(
```

Comment:
Does this imply that all our metrics emit code will be interacting with blocking mutexes?

Reply:
No, this is actually old code. It's been removed in #4431. I initially thought it would be necessary to lock the controller for reads, since GraphQL resolvers are async and could be pulling from stats simultaneously. But the `Controller` is already capable of parallel reads.